src/components/calcite-input/calcite-input.tsx

```diff
@@ -166,14 +196,26 @@ export class CalciteInput {
 
   componentWillUpdate() {
     this.calciteInputInput.emit({
-      element: this.childEl as HTMLInputElement,
+      element: this.childEl,
       value: this.value,
     });
+
+    this.determineClearable();
   }
 
   render() {
     const dir = getElementDir(this.el);
     const attributes = this.getAttributes();
+
+    const inputClearButton = (
+      <div
```

Comment:
I would wager that this is not accessible. Ie. in a screen-reader I would not even know about this element. This is apparently how the current browser-native clears are: 

> Of 9 browsers tested, six get a delete control and two of those also get a delete control on type="text". That means 4 of them get a delete control from type="search".
>
> Since screen readers do not announce this delete control (when it appears) and it can seemingly only be activated by pointer (I could not get it work in Voice Control), I don’t find it a compelling reason to use type="search". This does not even address target size, contrast, nor lack of accessible name.

—https://adrianroselli.com/2019/07/ignore-typesearch.html



Seems like if we make some tweaks we could substantially improve the screen reader experience for input fields, which is great.  We probably should use a button with `type=button`, allow devs to provide a label. Here's a discussion on ux stack exchange: https://ux.stackexchange.com/questions/125755/need-explanation-how-clear-input-button-accessibility-works

Reply:
Yeah that'd be a good improvement. Using "Escape" on a "clearable" input also works but to the point it's not announced that is a possible interaction anyway. 

Can update to use a button and accept a "clear button string" that is only used if "clearable" is requested as a prop (or, I guess, additionally for input types that would get a browser one by default - search, time, date).